### ISSUE TITLE
Blazor 6.0 content tracking issue

### ISSUE THREAD

**Issue by guardrex** [dotnet/AspNetCore.Docs#22045]
### Assignment currently undetermined

* [ ] **Blazor WASM native dependencies support** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#native-dependencies-support) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/host-and-deploy/webassembly.md) | I hit a little 🐛 while working on this, but it seems to be some kind of Emscripten problem with my ancient 10-year-old PC 🙈.
* [ ] **WebAssembly AoT Compilation with runtime relinking guidance** | [AOT Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#ahead-of-time-aot-compilation) | [Runtime Relinking Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#runtime-relinking) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/host-and-deploy/webassembly.md) | Reviewed (but not approved) by DR.
* [ ] **Persist prerendering state** | [Live 1](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/prerendering-and-integration?view=aspnetcore-6.0&pivots=server#preserve-prerendered-state) | [Edit 1](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/prerendering-and-integration.md) | [Live (Tag Helper)](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/preserve-component-state?view=aspnetcore-6.0&pivots=server) | [Edit 2](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/mvc/views/tag-helpers/built-in/preserve-component-state.md) | Approved by DR.
* [ ] **NRTs and compiler null-state checks** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/project-structure?view=aspnetcore-6.0#nullable-reference-types-nrts-and-net-compiler-null-state-static-analysis) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/project-structure.md) | There will be a final pass to update inline doc examples after 6.0 GA and hopefully 🤞 before the 🎁 *holiday break* ⛄.
* [ ] **Blazor 6.0 migration content** | [23607](https://github.com/dotnet/AspNetCore.Docs/pull/23607) | [Live](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/migration/50-to-60.md) | PR approved by Artak.

### Hao (and perhaps Barry)

* [ ] **Duende for Identity Server/AAD** | [23047](https://github.com/dotnet/AspNetCore.Docs/pull/23047) | See PR because this touched several spots.

### Safia

* [ ] **Add support for error boundaries in Blazor** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-6.0&pivots=server#error-boundaries) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/includes/handle-errors/error-boundaries.md)
* [ ] **Improve SVG support in Blazor** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#scalable-vector-graphics-svg-images) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)

### Tanay

* [ ] **Blazor Server: Support byte-array transfer in JS Interop** | [Call JS Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#byte-array-support) | [Call .NET Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-6.0#byte-array-support) | [Call JS Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md) | [Call .NET Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md)
* [ ] **Blazor Server: Remove size limitations imposed by SignalR message size** | [23739](https://github.com/dotnet/AspNetCore.Docs/pull/23739) & [23740](https://github.com/dotnet/AspNetCore.Docs/pull/23740)

### Mackinnon

* [ ] **Query string enhancements** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-6.0#query-strings) (+6 sub-sections) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/routing.md)
* [ ] **Support binding to select multiple, getting/setting the selection as an array** | [Live (Binding)](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/data-binding?view=aspnetcore-6.0#multiple-option-selection-with-input-elements) (+next section) | [Live (Forms)](https://docs.microsoft.com/en-us/aspnet/core/blazor/forms-validation?view=aspnetcore-6.0#multiple-option-selection-with-the-inputselect-component) | [Edit (Binding)](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/data-binding.md) | [Edit (Forms)](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/forms-validation.md)
* [ ] **Update Blazor head components** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content?view=aspnetcore-6.0) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/control-head-content.md)
* [ ] **Generate Angular and React components using Blazor** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#generate-angular-and-react-components) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)

### Steve

* [ ] **Render components from JS** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#render-razor-components-from-javascript) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)
* [ ] **Blazor custom elements** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#blazor-custom-elements) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)
* [ ] **Infer component generic types from ancestor components** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/templated-components?view=aspnetcore-6.0#infer-generic-types-based-on-ancestor-components) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/templated-components.md)
* [ ] **Dynamically rendered components (`DynamicComponent`)** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/dynamiccomponent?view=aspnetcore-6.0) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/dynamiccomponent.md)
* [ ] **Improved Blazor accessibility: `FocusOnNavigate`** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-6.0#focus-an-element-on-navigation) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/routing.md)
* [ ] **Support custom event args in Blazor** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-6.0#custom-event-arguments) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/event-handling.md)

### Pranav

* [ ] **Required parameters to blazor components** | [Live (scroll to end of section)](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#component-parameters) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)

### Javier

* [ ] **Collocate JS files with pages, views, and components** | [include](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/includes/js-collocation.md) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/includes/js-collocation.md) | The INCLUDE is displayed in the Blazor JS interop overview topic and the Blazor RCL topic. **More work needed to surface for RP/MVC. I've noted this in my tracking for more work, which I'll reach this week (Wednesday, 10/27). For more info on this work, see the PR at [23327](https://github.com/dotnet/AspNetCore.Docs/pull/23327).
* [ ] **JavaScript initializers** | [include](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/includes/js-initializers.md) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/includes/js-initializers.md) | Include displayed in the Blazor Fundamentals Startup topic and the JS interop overview.
* [ ] **.NET to JS Streaming Interop** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#stream-from-net-to-javascript) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md) | This and the next entry are in the *Call JS from .NET* topic SxS in two sections. This was done because "call" in this context is a bit muddy. Who is "calling" who given that these sections are focused on "streaming," not "calling," per se. You might decide that we need to move one or the other over to the *Call .NET from JS* topic. :ear:
* [ ] **Stream from JavaScript to .NET** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#stream-from-javascript-to-net) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md) | This and the previous entry are in the *Call JS from .NET* topic SxS in two sections. You might decide that we need to move one or the other over to the *Call .NET from JS* topic. :ear:
* [ ] **Allow generic type constraints** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#generic-type-parameter-support) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)
* [ ] **WASM deployment layout topic** | [Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly-deployment-layout?view=aspnetcore-6.0) | [Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/host-and-deploy/webassembly-deployment-layout.md) | ⚔️ *Sir blowdart* 🎯 is ok with note content on bypassing DLL-network restrictions.

### On HOLD or Coming Soon

* [ ] **Customize Blazor WASM packaging** 🛑 HOLD for PU/blog-supplied content. This may refer to a remark on the [JS collocate/initializers PR](https://github.com/dotnet/sdk/pull/19270).
* [ ] **Dynamically added root components** 🛑 HOLD for PU-supplied content.
* [ ] **Blazor hybrid desktop apps (.NET MAUI)** 🛑 HOLD for PU coordination offline.

### Dropped or not required

* **PageX and PageY** No docs required. This is covered by API docs.
* **Blazor Server: More control over circuits lifetime** CUT by PU.
* **Expose location changing event for `NavigationManager`** CUT by PU.
* **Drag & Drop** CUT by PU.
* **Input `ElementReference` exposed on relevant components** No docs required.
* **Specify `null` for `Action`/`RenderFragment` component params** No docs required.
* **Pause and resume Blazor applications** CUT by PU.
* **Blazor Server: Specific components for dealing with large binary data (files and media)** CUT by PU.
* **Blazor Server: Specific component for dealing with large amounts of text data (`InputTextArea`)** CUT by PU.
* **Launch DebugProxy from `blazorwasm` debug type in VS Code** No docs required.

## 6.0

Status | Docs Issue | Docs PR | Item | Notes
:---: | :---: | :---: | --- | ---
✔️ | [23522](https://github.com/dotnet/AspNetCore.Docs/issues/23522) | [23530](https://github.com/dotnet/AspNetCore.Docs/pull/23530) | Blazor WASM native dependencies support | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-2/#native-dependencies-support-for-blazor-webassembly-apps). Hit a 🐛 on the PR, but it was probably just a point-in-time thing with RC2. I'll check the scenario at GA.
✔️ | [23402](https://github.com/dotnet/AspNetCore.Docs/issues/23402) | [23411](https://github.com/dotnet/AspNetCore.Docs/pull/23411)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly-deployment-layout?view=aspnetcore-6.0)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/host-and-deploy/webassembly-deployment-layout.md) | WASM deployment layout topic | ⚔️ *Sir blowdart* 🎯 and I discussed the general *bypass-DLL-network-restrictions-at-your-own-risk* NOTE offline, and the *DRAFT* content should suffice until final review+approval prior to GA.
✔️ | [23294](https://github.com/dotnet/AspNetCore.Docs/issues/23294) | [23313](https://github.com/dotnet/AspNetCore.Docs/pull/23313)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#render-razor-components-from-javascript)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md) | Render components from JS | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#render-blazor-components-from-javascript)
✔️ | [23295](https://github.com/dotnet/AspNetCore.Docs/issues/23295) | [23314](https://github.com/dotnet/AspNetCore.Docs/pull/23314)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#blazor-custom-elements)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md) | Blazor custom elements | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#blazor-custom-elements)
✔️ | [23297](https://github.com/dotnet/AspNetCore.Docs/issues/23297) | [23315](https://github.com/dotnet/AspNetCore.Docs/pull/23315)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#generate-angular-and-react-components)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md) | Generate Angular and React components using Blazor | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#generate-angular-and-react-components-using-blazor)
✔️ | [23298](https://github.com/dotnet/AspNetCore.Docs/issues/23298) | ❌ | PageX and PageY | **No docs required.** This is covered by API docs. [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#pagex-and-pagey-in-mouseeventargs)
✔️ | [23299](https://github.com/dotnet/AspNetCore.Docs/issues/23299) | [23327](https://github.com/dotnet/AspNetCore.Docs/pull/23327)<br>[include](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/includes/js-collocation.md)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/includes/js-collocation.md)<br>See NOTE :point_right: | Collocate JS files with pages, views, and components | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#collocate-javascript-files-with-pages-views-and-components) [PU PR](https://github.com/dotnet/sdk/pull/19270); NOTE: Include displayed in Blazor JS interop overview topic and Blazor RCL topic. More work needed to surface for RP/MVC. See the docs PR for remarks, and I've noted this below in *Non-Blazor (but related) features* area for more work.
✔️ | [23300](https://github.com/dotnet/AspNetCore.Docs/issues/23300) | [23331](https://github.com/dotnet/AspNetCore.Docs/pull/23331)<br>[include](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/includes/js-initializers.md)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/includes/js-initializers.md)<br>See NOTE :point_right: | JavaScript initializers | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#javascript-initializers) [PU PR](https://github.com/dotnet/sdk/pull/19270); NOTE: Include displayed in Blazor Fundamentals Startup topic and JS interop overview.
❓ | &mdash; | &mdash; | Customize Blazor WASM packaging | I think this might be a duplicate entry for [JS initializers](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/#javascript-initializers) and [deployment layouts](https://docs.microsoft.com/aspnet/core/blazor/host-and-deploy/webassembly-deployment-layout).
✔️ | [23039](https://github.com/dotnet/AspNetCore.Docs/issues/23039) | [23046](https://github.com/dotnet/AspNetCore.Docs/pull/23046)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#stream-from-net-to-javascript)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md)<br>See NOTE :point_right: | .NET to JS Streaming Interop | [PU PR](https://github.com/dotnet/aspnetcore/pull/34817); NOTE: This and the next entry are in the *Call JS from .NET* topic SxS in two sections. This was done because "call" in this context is a bit muddy. Who is "calling" who given that these sections are focused on "streaming," not "calling," per se. You might decide that we need to move one or the other over to the *Call .NET from JS* topic. :ear:
✔️ | [23013](https://github.com/dotnet/AspNetCore.Docs/issues/23013) | [23014](https://github.com/dotnet/AspNetCore.Docs/pull/23014)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#stream-from-javascript-to-net)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md)<br>See NOTE :point_right: | Stream from JavaScript to .NET | [Blog Post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-7/#blazor-streaming-interop-from-javascript-to-net) [PU PR](https://github.com/dotnet/aspnetcore/pull/33491); NOTE: This and the previous entry are in the *Call JS from .NET* topic SxS in two sections. This was done because "call" in this context is a bit muddy. Who is "calling" who given that these sections are focused on "streaming," not "calling," per se. You might decide that we need to move one or the other over to the *Call .NET from JS* topic. :ear:
✔️ | [22355](https://github.com/dotnet/AspNetCore.Docs/issues/22355) | [23047](https://github.com/dotnet/AspNetCore.Docs/pull/23047)<br>See PR because this touched several spots.<br>Already signed off by Barry and Hao. | Duende for Identity Server/AAD | [PU Issue](https://github.com/dotnet/aspnetcore/issues/32881) Applies to the entire doc set (Blazor + non-Blazor). See: [32881](https://github.com/dotnet/aspnetcore/issues/32881)
✔️ | [22863](https://github.com/dotnet/AspNetCore.Docs/issues/22863) | [23033](https://github.com/dotnet/AspNetCore.Docs/pull/23033)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/signalr?view=aspnetcore-6.0&pivots=server#long-polling)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/signalr.md)<br>PR approved by Tanay. | Using Long Polling for Blazor Server | [PU Issue](https://github.com/dotnet/aspnetcore/pull/34644) [PU PR](https://github.com/dotnet/aspnetcore/pull/34644)
✔️ | [22689](https://github.com/dotnet/AspNetCore.Docs/issues/22689) | [23000](https://github.com/dotnet/AspNetCore.Docs/pull/23000)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-6.0#query-strings) (+6 sub-sections)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/routing.md)<br>PR approved by Mackinnon. | Query string enhancements | [PU Issue](https://github.com/dotnet/aspnetcore/issues/33338) [PU PR 1](https://github.com/dotnet/aspnetcore/issues/33840) [PU Design for building](https://github.com/dotnet/aspnetcore/issues/34115#issuecomment-882587986) [PU PR 2](https://github.com/dotnet/aspnetcore/pull/34813)
🛑<br>**HOLD** until RC2 release for PU-supplied content | [22690](https://github.com/dotnet/AspNetCore.Docs/issues/22690) | [23032](https://github.com/dotnet/AspNetCore.Docs/pull/23032)&dagger; | Dynamically added root components | [PU Issue](https://github.com/dotnet/aspnetcore/issues/27574) [PU PR](https://github.com/dotnet/aspnetcore/pull/34232) &dagger;Had to close due to need for PU input. Check back with SS after RC2 releases.
✔️ | [22440](https://github.com/dotnet/AspNetCore.Docs/issues/22440) | [23065](https://github.com/dotnet/AspNetCore.Docs/pull/23065)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-6.0&pivots=windows#net-hot-reload)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/tooling.md)<br>Reviewed (but not approved) by Dan. | Hot reload for Blazor | **Feature Still Under Development** [PU Issue Search](https://github.com/dotnet/aspnetcore/issues?q=is%3Aissue+is%3Aopen+blazor+hot+reload) [PU Issue](https://github.com/dotnet/aspnetcore/issues/18486) [Blog Post 3](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#initial-net-hot-reload-support) [Blog Post 5](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-5/#net-hot-reload-updates-for-dotnet-watch)
✔️ | [22442](https://github.com/dotnet/AspNetCore.Docs/issues/22442) | [23065](https://github.com/dotnet/AspNetCore.Docs/pull/23065)<br>[AOT Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#ahead-of-time-aot-compilation)<br>[RR Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-6.0#runtime-relinking)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/host-and-deploy/webassembly.md)<br>Reviewed (but not approved) by Dan. | WebAssembly AoT Compilation with runtime relinking guidance | [PU Issue](https://github.com/dotnet/aspnetcore/issues/5466) [Blog Post 4](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#blazor-webassembly-ahead-of-time-aot-compilation) [Blog Post 5](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-5/#reduced-blazor-webassembly-download-size-with-runtime-relinking)
🛑<br>**HOLD** for PU coordination offline | [22441](https://github.com/dotnet/AspNetCore.Docs/issues/22441) | &mdash; | Blazor hybrid desktop apps (.NET MAUI) | [PU Issue](https://github.com/dotnet/aspnetcore/issues/27217) [Blog Post 3](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/#blazorwebview-controls-for-wpf-windows-forms) [Blog Post 4](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#net-maui-blazor-apps)
✔️ | [22443](https://github.com/dotnet/AspNetCore.Docs/issues/22443) | [22683*](https://github.com/dotnet/AspNetCore.Docs/pull/22683)<br>See NOTE :point_right: | Launch DebugProxy from `blazorwasm` debug type in VS Code | [PU Issue](https://github.com/dotnet/aspnetcore/issues/22587); NOTE: \*[Closed](https://github.com/dotnet/AspNetCore.Docs/pull/22683#discussion_r668272519) as **no docs required** per Capt. Safia 🎖️.
✔️ | [21769](https://github.com/dotnet/AspNetCore.Docs/issues/21769) | [21846](https://github.com/dotnet/AspNetCore.Docs/pull/21846)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/templated-components?view=aspnetcore-6.0#infer-generic-types-based-on-ancestor-components)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/templated-components.md) | Infer component generic types from ancestor components | [PU Issue](https://github.com/dotnet/aspnetcore/issues/29349) [PU PR](https://github.com/dotnet/aspnetcore/pull/31800) [Blog Post 2](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-2/#infer-component-generic-types-from-ancestor-components)
✔️ | [22444](https://github.com/dotnet/AspNetCore.Docs/issues/22444) | ❌ | Blazor Server: More control over circuits lifetime | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/30287)
✔️ | [22444](https://github.com/dotnet/AspNetCore.Docs/issues/22444) | [22718](https://github.com/dotnet/AspNetCore.Docs/pull/22718)<br>[Call JS Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-6.0#byte-array-support)<br>[Call .NET Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-6.0#byte-array-support)<br>[Call JS Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-javascript-from-dotnet.md)<br>[Call .NET Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md)<br>PR approved by Tanay.  | Blazor Server: Support byte-array transfer in JS Interop | [PU Issue](https://github.com/dotnet/aspnetcore/issues/21877) [PU PR](https://github.com/dotnet/aspnetcore/pull/33015) [Announcement](https://github.com/dotnet/announcements/issues/187)
✔️ | [22444](https://github.com/dotnet/AspNetCore.Docs/issues/22444) | [23031](https://github.com/dotnet/AspNetCore.Docs/pull/23031)<br>IIRC, approved by Tanay in issue discussion. | Blazor Server: Remove size limitations imposed by SignalR message size | [PU Issue](https://github.com/dotnet/aspnetcore/issues/30289)
✔️ | [22444](https://github.com/dotnet/AspNetCore.Docs/issues/22444) | ❌ | Blazor Server: Specific components for dealing with large binary data (files and media) | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/30290)
✔️ | [22444](https://github.com/dotnet/AspNetCore.Docs/issues/22444) | ❌ | Blazor Server: Specific component for dealing with large amounts of text data (`InputTextArea`) | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/30291)
✔️ | [21763](https://github.com/dotnet/AspNetCore.Docs/issues/21763) | [21755](https://github.com/dotnet/AspNetCore.Docs/pull/21755)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-6.0#custom-event-arguments)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/event-handling.md) | Support custom event args in Blazor | [PU Issue](https://github.com/dotnet/aspnetcore/issues/17552) [Blog Post 2](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-2/#support-for-custom-event-arguments-in-blazor)
✔️ | [22445](https://github.com/dotnet/AspNetCore.Docs/issues/22445) | [22696](https://github.com/dotnet/AspNetCore.Docs/pull/22696)<br>[Live (scroll to end of section)](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#component-parameters)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md) | Required parameters to blazor components | [PU Issue](https://github.com/dotnet/aspnetcore/issues/11815) [PU PR](https://github.com/dotnet/aspnetcore/pull/33384) [Blog Post 6](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-6/#required-blazor-component-parameters)
✔️ | [17351](https://github.com/dotnet/AspNetCore.Docs/issues/17351) | [22697](https://github.com/dotnet/AspNetCore.Docs/pull/22697)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#generic-type-parameter-support)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md) | Allow generic type constraints | [PU Issue](https://github.com/dotnet/aspnetcore/issues/8433) [Blog Post 4](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#generic-type-constraints-in-razor)
✔️ | [22446](https://github.com/dotnet/AspNetCore.Docs/issues/22446) | [22688](https://github.com/dotnet/AspNetCore.Docs/pull/22688)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-6.0&pivots=server#error-boundaries)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/includes/handle-errors/error-boundaries.md)<br>PR approved by Safia. | Add support for error boundaries in Blazor | [PU Issue](https://github.com/dotnet/aspnetcore/issues/26953) [Blog Post 4](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/#blazor-error-boundaries)
✔️ | [22447](https://github.com/dotnet/AspNetCore.Docs/issues/22447) | ❌ | Pause and resume Blazor applications | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/27576)
✔️ | [22448](https://github.com/dotnet/AspNetCore.Docs/issues/22448) | [22770](https://github.com/dotnet/AspNetCore.Docs/pull/22770)<br>[Live (Binding)](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/data-binding?view=aspnetcore-6.0#multiple-option-selection-with-input-elements) (+next section)<br>[Live (Forms)](https://docs.microsoft.com/en-us/aspnet/core/blazor/forms-validation?view=aspnetcore-6.0#multiple-option-selection-with-the-inputselect-component)<br>[Edit (Binding)](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/data-binding.md)<br>[Edit (Forms)](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/forms-validation.md)<br>PR approved by Mackinnon. | Support binding to select multiple, getting/setting the selection as an array | [PU Issue](https://github.com/dotnet/aspnetcore/issues/5519) [PU PR](https://github.com/dotnet/aspnetcore/pull/33950)
✔️ | [22449](https://github.com/dotnet/AspNetCore.Docs/issues/22449) | ❌ | Expose location changing event for `NavigationManager` | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/14962)
✔️ | [22005](https://github.com/dotnet/AspNetCore.Docs/issues/22005) | [22984](https://github.com/dotnet/AspNetCore.Docs/pull/22984)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/dynamiccomponent?view=aspnetcore-6.0)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/dynamiccomponent.md)<br>PR approved by Dave Brock. | Dynamically rendered components (`DynamicComponent`) | [PU Issue](https://github.com/dotnet/aspnetcore/issues/26781) [Blog Post 1](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-1/#dynamiccomponent) [PU PR](https://github.com/dotnet/aspnetcore/pull/28082) [DB Blog Post](https://www.daveabrock.com/2021/04/08/blazor-dynamic-component/)
✔️ | [20319](https://github.com/dotnet/AspNetCore.Docs/issues/20319) | [22929](https://github.com/dotnet/AspNetCore.Docs/pull/22929)<br>[22931](https://github.com/dotnet/AspNetCore.Docs/pull/22931)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/control-head-content?view=aspnetcore-6.0)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/control-head-content.md)<br>PR approved by Mackinnon. | Update Blazor head components. | [PU Issue](https://github.com/dotnet/aspnetcore/issues/25705) [PU PR 1](https://github.com/dotnet/aspnetcore/pull/34218) [PU PR 2](https://github.com/dotnet/aspnetcore/pull/34503)
✔️ | &mdash; | [23364](https://github.com/dotnet/AspNetCore.Docs/pull/23364) | Prerendering and integration topic updates | These updates include control of head content for the scenarios described in the topic (i.e., the *third step* of updates on the *Update Blazor head components* 🤠 **_Texas Three-step_**&trade; 🕺💃 called out on those issues/PRs :point_up:). 
✔️ | [22450](https://github.com/dotnet/AspNetCore.Docs/issues/22450) | [22717](https://github.com/dotnet/AspNetCore.Docs/pull/22717)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#scalable-vector-graphics-svg-images)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/index.md)<br>PR approved by Safia. | Improve SVG support in Blazor | [PU I1](https://github.com/dotnet/aspnetcore/issues/18271) [PU I2](https://github.com/dotnet/aspnetcore/issues/30973) [PU PR](https://github.com/dotnet/aspnetcore/pull/32417) [Blog Post 5](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-5/#support-razor-syntax-in-svg-foreignobject-elements)
✔️ | [21992](https://github.com/dotnet/AspNetCore.Docs/issues/21992) | ❌ | Drag & Drop | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/18754)
✔️ | [21771](https://github.com/dotnet/AspNetCore.Docs/issues/21771) | [22732](https://github.com/dotnet/AspNetCore.Docs/pull/22732) (First added PR)<br>[23621](https://github.com/dotnet/AspNetCore.Docs/pull/23621) (Update PR)<br>[Live 1](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/prerendering-and-integration?view=aspnetcore-6.0&pivots=server#preserve-prerendered-state)<br>[Edit 1](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/components/prerendering-and-integration.md)<br>[Live (Tag Helper)](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/preserve-component-state?view=aspnetcore-6.0&pivots=server)<br>[Edit 2](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/mvc/views/tag-helpers/built-in/preserve-component-state.md)<br>PR approved by Dan. | Persist prerendering state | [Blog Post 2](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-2/#preserve-prerendered-state-in-blazor-apps)
✔️ | [22451](https://github.com/dotnet/AspNetCore.Docs/issues/22451) | ❌  | Input `ElementReference` exposed on relevant components | **No docs required.** [Blog Post 1](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-1/#input-elementreference-exposed-on-relevant-components)
✔️ | ❌ | ❌  | Specify `null` for `Action`/`RenderFragment` component params | **No docs required.** [Blog Post 5](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-5/#specify-null-for-action-and-renderfragment-component-parameters)
✔️ | [23603](https://github.com/dotnet/AspNetCore.Docs/issues/23603) | [23607](https://github.com/dotnet/AspNetCore.Docs/pull/23607)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/migration/50-to-60.md) | Blazor 6.0 migration content | Approved by Artak.
✔️ | [22734](https://github.com/dotnet/AspNetCore.Docs/issues/22734) | [22735](https://github.com/dotnet/AspNetCore.Docs/pull/22735)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-6.0#focus-an-element-on-navigation)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/fundamentals/routing.md) | Improved Blazor accessibility: `FocusOnNavigate` | [Blog Post 6](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-6/#improved-blazor-accessibility)
✔️ | [23392](https://github.com/dotnet/AspNetCore.Docs/issues/23392) | [23504](https://github.com/dotnet/AspNetCore.Docs/pull/23504)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/blazor/project-structure?view=aspnetcore-6.0#nullable-reference-types-nrts-and-net-compiler-null-state-static-analysis)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/project-structure.md) | NRTs and compiler null-state checks | I've decided to notify readers via the *Project Structure* topic, where the Blazor templates are discussed. The updates to doc examples will occur transparently because C# language features are beyond the scope of Blazor docs ... we just want the doc examples updated to embrace NRTs and null-state checks. The code example work will be tracked by this issue (6.0 GA table :point_down:). The PR for to resolve this item is just to call it out in the text.

## RC1

* [x] #23482 Resolved by #23490 and #23491.
* [x] Minimal API updates: Sweep for `Startup.cs`, `Startup.ConfigureServices`, `Startup.Configure`, `Program.Main` (and `Main` methods) in Blazor  guidance. #23483
* [x] Check on the project structure topic vis-a-vis *Additional content for Blazor Server `_Layout.cshtml`*: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/#blazor-templates-updated-to-set-page-title **_Checked ... no updates needed._**
* [x] Remove the *new state preservation feature* language per https://github.com/dotnet/AspNetCore.Docs/pull/22732: Search on "6.0" in the Blazor node. https://github.com/dotnet/AspNetCore.Docs/pull/23308
* [x] Snippet sample apps and EFCore-Server sample app to RC1. https://github.com/dotnet/AspNetCore.Docs/pull/23343 &mdash; https://github.com/dotnet/AspNetCore.Docs/pull/23343
* [x] SignalR-Blazor tutorial sample apps update to RC1. https://github.com/dotnet/AspNetCore.Docs/pull/23397
* [x] Todo List tutorial update to RC1. https://github.com/dotnet/AspNetCore.Docs/pull/23397
* [x] #20951 &mdash; Confirm that `PreferExactMatches` INCLUDES content/versioning is good and that it's removed from sample apps. **_Checked ... no updates needed._**

## RC2

* [x] Snippet sample apps and EFCore-Server sample app to RC2. Updates are on #23518.
* [x] SignalR-Blazor tutorial sample apps update to RC2. Updates are on #23518.
* [x] Check on API renames :point_right: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-2/#breaking-changes-apis-renames. **_Checked ... no updates needed._**

## PU roadmap, announcements, and blog posts

* https://github.com/dotnet/aspnetcore/issues/27883
* https://github.com/aspnet/Announcements/issues?q=is%3Aissue+label%3A6.0.0
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-1/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-2/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-3/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-4/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-5/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-6/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-7/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-1/
* https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-rc-2/
* https://devblogs.microsoft.com/dotnet/announcing-asp-net-core-in-net-6/

## 6.0 Post GA

Status | Docs Issue | Docs PR | Subject
:---: | :---: | :---: | ---
✔️ | &nbsp; | &nbsp; | For #23668, update the native dep cross-link the *Hosting Model* topic. **_Done!_** #23668
✔️ | &nbsp; | &nbsp; | After the [Native Deps PR (dotnet/AspNetCore.Docs #23668)](https://github.com/dotnet/AspNetCore.Docs/pull/23668) merges and goes live: Open a follow-up issue on HOLD to convert preview language for SkiaSharp when SkiaSharp reaches GA. **_Done!_** #23904
✔️ | #23825 | #23831 | *Hot Reload* guidance move follow-up cross-link update for *What's New* topic.
Tracked for final work 🛠️ by 24491 | [24491](https://github.com/dotnet/AspNetCore.Docs/issues/24491) | #23842 | Review and update *What's New* content for Blazor, which will be located [here](https://docs.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-6.0) after publication. [Original content](https://github.com/dotnet/AspNetCore.Docs/issues/22852#issuecomment-952811181). The last three items require content and removal of HTML comment markers.
&nbsp; | #23064 | &nbsp; | Move new 6.0 example code block snippets into the Blazor snippet sample apps (include NRT work on these examples).
✔️ | #23063 | #23842 | Add 6.0 Blazor API doc cross-links
✔️ | #21897 | #23811 | Snippet sample apps and EFCore-Blazor Server final updates to release bits.
💥 | #21899 | &nbsp; | Blazor-SignalR tutorial confirm/update VS UI guidance (Windows and macOS) **_Waiting on Aquent Mac_**
✔️ | #21898 | #23365 | Policy hashes. I think this is resolved by https://github.com/dotnet/AspNetCore.Docs/pull/23365. If not, the readership will **_sharpen their beaks and claws for me!_** 🦅😨 
✔️ | &nbsp; | #23365 | Merge this PR | &nbsp;
✔️ | &nbsp; | &nbsp; | Check/update the API doc cross-links in the *Blazor Hub endpoint route configuration* section of *Fundamentals* > *SignalR* topic. **Content OK :+1: ... no updates needed.**
✔️ | &nbsp; | #23813 | Cross-link the Minimal API doc in the main doc set into the Blazor docs.
✔️ | &nbsp; | #23813 | **GA** Revert "`*preview release*`" language for 6.0 from multiple topics where it appears. Also search and destroy for anything mentioning "6.0 preview".
✔️ | &nbsp; | #23813 | Update doc dates to GA date
✔️ | &nbsp; | #23813 | Update VS 2019 (the date) to VS 2022.
✔️ | #23620 | #23504 | NRT and compiler null-state static analysis updates. Snippet sample apps are done, sans one EF Core-Blazor Server NRT 😈. Guidance appears at [Nullable reference types (NRTs) and .NET compiler null-state static analysis](https://docs.microsoft.com/en-us/aspnet/core/blazor/project-structure?view=aspnetcore-6.0#nullable-reference-types-nrts-and-net-compiler-null-state-static-analysis). The remaining work to address inline doc examples hopefully will occur before the 🎁 *holiday vacation* ⛄ and no later than 22Q1. The issue is tracked by the Blazor.Docs project, so it doesn't block closing out this 6.0 tracking issue.
✔️ | #23864 | &nbsp; | Open a followup issue for "\* In Visual Studio 2022 GA (17.0), Hot Reload is only supported when running without the debugger." when VS can handle it. `aspnetcore/test/hot-reload.md` **_Done!_** ... and marked on HOLD on the Blazor.Docs project so that it won't get lost.
💥 | &nbsp; | &nbsp; | **PROBLEM ATM**: My iMac is so old that it can't take macOS 10.14. Therefore, it can't take VS for Mac 2022. Therefore, I can't update docs for UI changes. I'm waiting on Aquent to send me a late-model Mac. This entry will be addressed as soon as that Mac arrives. 🚚<br><br>VS for Mac 2022 is still in preview for 6.0. Search for "preview" to isolate and update the guidance. It's probably best to keep the text in the docs where it cross-links to content on how to use the *Preview* channel and just comment it out. This is going to be recurring guidance each time VS for Mac goes into preview for an upcoming release, and it will be simpler to keep the guidance there in an *unrendered state* between releases. Note that this update includes Blazor node docs and the Blazor-SignalR tutorial.

## Non-Blazor (but related) features

Status | Docs Issue | Docs PR | Subject | Source
:---: | :---: | :---: | --- | ---
✔️ | [23120](https://github.com/dotnet/AspNetCore.Docs/issues/23120) | [23389](https://github.com/dotnet/AspNetCore.Docs/pull/23389)<br>[Live](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/?view=aspnetcore-6.0&tabs=visual-studio#css-isolation)<br>[Edit](https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/includes/css-isolation.md) | RP/MVC CSS isolation| [PU Issue](https://github.com/dotnet/aspnetcore/issues/25713) [PU PR](https://github.com/dotnet/aspnetcore/pull/29899) [Blog Post 2](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-6-preview-2/#css-isolation-for-mvc-views-and-razor-pages)
✔️ | ❌ | ❌ | Single-file publishing | **CUT by PU** [PU Issue](https://github.com/dotnet/aspnetcore/issues/27888)

**Comment by fiyazbinhasan** [comment]
Hey @guardrex! I can see some cool stuff coming in Blazor 6.0 docs. Do you need an extra hand? Let me know how you are planning to resolve these. Thanks 😺

**Comment by guardrex** [comment]
Yes, @fiyazbinhasan ... that would be great to have help with the updates. They're on hold right at the moment for two reasons ...

* We've been burned a few times in the past documenting upcoming release features, particularly the large features (new topics usually) during preview due to engineering churn. We document something well ... the engineers take feedback and think things over and then decide to change the bits ... the *new* docs are then stale and must be revisited and updated. If the changes are small, no problem. Sometimes tho, the changes are *LARGE*. It's time-consuming and costly 💰 when this happens, so we generally hold off until RC1 (or close to RC1) before documenting most upcoming release features. We're still a couple of months away from RC1. Historically, there's been plenty of time between RC1 and GA to get everything documented, too. There's usually at least two full months, often three.
* I'm not aware yet on a few of these if the PU wants to provide content, possibly code, for the doc updates. For most of these, text and code is readily available from the PU issue/PR and DR's preview blog posts. For most of these, the PU will be quite happy not to have to provide anything ... *indeed, the less they hear about docs the better* 🙈😆.

When the time comes to work on these, the plan is to use the individual tracking issues (first column :point_up:) to decide where the content will live (e.g., section of a topic or new topic, titling, metadata, etc.) and what sort of example code would be good. To speed up working example code in, what we'll probably do is place code directly into the topics and then I'll circle around after GA and move that code into the snippet sample apps.

I'll ping u back here when I have more info on the plan for these.

**Comment by fiyazbinhasan** [comment]
> I'll ping u back here when I have more info on the plan for these.

Sure thing @guardrex! 

Thanks for explaining everything I need to know. One thing though, how do you type all of these so fast? :scream_cat:

**Comment by guardrex** [comment]
They installed an electric shock feature on my PC. If I don't answer an issue comment within 37 seconds, **_they hit me 240V!_** 😝

**Comment by guardrex** [comment]
@fiyazbinhasan ... I haven't forgotten about your offer to help. Currently, I'm working through a small batch of mostly straightforward issues that aren't taking long to resolve. I'm almost done with them. I'll have this subset addressed within a couple of weeks. We're in good shape right now with the workload. I'll let you know if help is needed and/or I come to larger issues that require a lot more work.

**Comment by fiyazbinhasan** [comment]
Aye, aye, captain! 😄

**Comment by TanayParikh** [comment]
Hey @guardrex, just wanted to mention we don't need docs for `Blazor Server: Specific component for dealing with large amounts of text data (InputTextArea)` as we've decided not to add it to the core framework (but instead put up a sample in aspnet/ASPLabs).
